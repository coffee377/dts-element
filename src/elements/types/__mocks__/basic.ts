export class BasicType {

  public parameters: any;

  constructor(parameters: any) {
    this.parameters = parameters;
  }

  public _emit(): string {
    return `[BasicType ${this.parameters.name}]`;
  }

}
