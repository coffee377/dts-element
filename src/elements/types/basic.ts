import {AnyElement} from '../../element';
import {Type} from '../type';

export interface IBasicTypeRequiredParameters {
  name: string;
}

// tslint:disable-next-line no-empty-interface
export interface IBasicTypeOptionalParameters {}

export class BasicType extends Type<IBasicTypeRequiredParameters, IBasicTypeOptionalParameters> {

  public get default_parameters(): IBasicTypeOptionalParameters {
    return {};
  }

  public _emit(_container: AnyElement | null): string {
    return this.parameters.name;
  }

}
