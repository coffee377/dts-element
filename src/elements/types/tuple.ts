import {Container} from '../../collections';
import {emit_elements} from '../../helpers/emit-elements';
import {Type} from '../type';

export interface ITupleTypeRequiredParameters {
  owneds: Type[];
}

// tslint:disable-next-line no-empty-interface
export interface ITupleTypeOptionalParameters {}

export class TupleType extends Type<ITupleTypeRequiredParameters, ITupleTypeOptionalParameters> {

  public get default_parameters(): ITupleTypeOptionalParameters {
    return {};
  }

  public _emit(container: Container): string {
    return `[${emit_elements(this.parameters.owneds, container, ', ')}]`;
  }

}
