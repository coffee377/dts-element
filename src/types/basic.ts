import {AnyElement} from '../element';
import {Type} from '../type';

export interface IBasicRequiredParameters {
  name: string;
}

// tslint:disable-next-line no-empty-interface
export interface IBasicOptionalParameters {}

export class BasicType extends Type<IBasicRequiredParameters, IBasicOptionalParameters> {

  public get default_parameters(): IBasicOptionalParameters {
    return {};
  }

  public _emit(_container: AnyElement): string {
    return this.parameters.name;
  }

}
