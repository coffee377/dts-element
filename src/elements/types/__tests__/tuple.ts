jest.unmock('../tuple');

import {BasicType} from '../basic';
import {TupleType} from '../tuple';

describe('#emit()', () => {
  it('should return correctly', () => {
    expect(new TupleType({owneds: [
      new BasicType({name: 'A'}),
      new BasicType({name: 'B'}),
    ]}).emit()).toMatchSnapshot();
  });
});
