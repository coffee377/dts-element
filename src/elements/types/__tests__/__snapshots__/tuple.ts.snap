// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#emit() should return correctly 1`] = `"[[emit-elements [[BasicType A], [BasicType B]]]]"`;
